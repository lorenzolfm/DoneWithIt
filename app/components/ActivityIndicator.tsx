import React from 'react';
import LottieView from 'lottie-react-native';
import { StyleSheet, View } from 'react-native';

type Props = { visible: boolean }

export const ActivityIndicator = ({ visible = false }: Props) => {
  if (!visible) return null;

  return (
    <View style={styles.overlay}>
      <LottieView
        autoPlay
        loop
        source={require('../assets/animations/loader.json')}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  overlay:  {
    position: 'absolute',
    backgroundColor: 'white',
    height: '100%',
    width: '100%',
    zIndex: 1,
    opacity: 0.8,
  }
})
