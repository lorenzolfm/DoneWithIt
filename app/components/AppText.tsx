import React from 'react';
import { Text } from 'react-native';

import defaultStyles from '../config/styles';

type AppTextProps = {
  children?: React.ReactNode,
  style?: object,
  [otherProp:string]: any,
};
export const AppText = ({ children, style, ...otherProps }: AppTextProps) => {
  return (
    <Text style={[defaultStyles.text, style]} {...otherProps}>{children}</Text>
  );
};
