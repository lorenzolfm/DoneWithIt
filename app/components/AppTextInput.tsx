import React from 'react';
import { StyleSheet, TextInput, View } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import defaultStyles from '../config/styles';

type Props = {
  icon?: any,
  placeholder?: string,
  [otherProp:string]: any,
};
export const AppTextInput = ({ icon, placeholder, ...otherProps }: Props) => {
  return (
    <View style={styles.container}>
      {icon && <MaterialCommunityIcons name={icon} size={20} color={defaultStyles.colors.medium} style={styles.icon} />}
      <TextInput style={defaultStyles.text} placeholder={placeholder} {...otherProps}/>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: defaultStyles.colors.light,
    borderRadius: 25,
    flexDirection: 'row',
    width: '100%',
    padding: 15,
    marginVertical: 10,
    alignItems: 'center',
  },
  icon: {
    marginRight: 10,
  }
});
