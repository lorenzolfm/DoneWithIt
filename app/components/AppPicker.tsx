import React, { useState } from 'react';
import { StyleSheet, View, TouchableWithoutFeedback, Modal, Button, FlatList } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import defaultStyles from '../config/styles';
import { AppText } from './AppText';
import { Screen } from './Screen';

import { Category } from '../types';
import { PickerItem } from './PickerItem';

type Props = {
  icon?: any,
  placeholder?: string,
  items?: Category[],
  width?: number | string,
  selectedItem: Category,
  onSelectItem: Function,
};
export const AppPicker = ({ icon, placeholder, items, width = '100%', selectedItem, onSelectItem }: Props) => {
  const [modalVisible, setModalVisible] = useState(false);

  return (
    <>
      <TouchableWithoutFeedback onPress={() => setModalVisible(true)}>
        <View style={[styles.container, { width }]}>
          {icon && <MaterialCommunityIcons name={icon} size={20} color={defaultStyles.colors.medium} style={styles.icon} />}
          {selectedItem ?
            <AppText style={styles.text}>{selectedItem.label}</AppText> :
            <AppText style={styles.placeholder}>{placeholder}</AppText>
          }
          <MaterialCommunityIcons
            name='chevron-down'
            size={20}
            color={defaultStyles.colors.medium}
          />
        </View>
      </TouchableWithoutFeedback>
      <Modal visible={modalVisible} animationType="slide">
        <Screen>
          <Button title="Close" onPress={() => setModalVisible(false)} />
          <FlatList
            data={items}
            keyExtractor={item => item.value.toString()}
            renderItem={
              ({ item }) =>
                <PickerItem label={item.label} onPress={() => { setModalVisible(false); onSelectItem(item) }} />
            }
          />
        </Screen>
      </Modal>
    </>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: defaultStyles.colors.light,
    borderRadius: 25,
    flexDirection: 'row',
    width: '100%',
    padding: 15,
    marginVertical: 10,
    alignItems: 'center',
  },
  icon: {
    marginRight: 10,
  },
  text: {
    flex: 1,
  },
  placeholder: {
    flex: 1,
    color: defaultStyles.colors.medium,
  }
});
