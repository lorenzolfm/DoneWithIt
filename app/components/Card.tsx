import React from 'react';
import { Image, StyleSheet, View } from "react-native";
import { TouchableWithoutFeedback } from 'react-native-gesture-handler';

import defaultStyles from '../config/styles';
import { AppText } from "./AppText";

type CardProps = {
  title: string,
  subtitle: string,
  imageUrl: string,
  onPress: () => void,
}

export const Card = ({ title, subtitle, imageUrl, onPress }: CardProps) => {
  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <View style={styles.card}>
        <Image source={{uri: imageUrl}} style={styles.image}/>
        <View style={styles.detailsContainer}>
          <AppText style={styles.title} numberOfLines={1}>{title}</AppText>
          <AppText style={styles.subtitle} numberOfLines={2}>{subtitle}</AppText>
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

const styles = StyleSheet.create({
  card: {
    borderRadius: 15,
    backgroundColor: defaultStyles.colors.white,
    marginBottom: 20,
    overflow: 'hidden',
  },
  detailsContainer: { padding: 20 },
  image: {
    width: '100%',
    height: 200,
  },
  title: {
    marginBottom: 10,
  },
  subtitle: {
    color: defaultStyles.colors.secondary,
    fontWeight: 'bold',
  }
});
