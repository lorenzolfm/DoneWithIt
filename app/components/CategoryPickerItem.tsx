import React from 'react';
import { GestureResponderEvent, StyleSheet, View } from 'react-native';

import { AppText } from './AppText';
import { Icon } from './Icon';

type Props = {
  item: any
  onPress: (event: GestureResponderEvent) => void,
}

export const CategoryPickerItem = ({ item, onPress }: Props) => {
  return (
    <View style={styles.container}>
      <Icon backgroundColor={item.backgroundColor} name={item.icon} size={80} />
      <AppText style={styles.label}>{item.label}</AppText>
    </View>
  )
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 20,
    paddingVertical: 15,
    alignItems: 'center',
    width: '33%',
  },
  label: {
    marginTop: 5,
    textAlign: 'center',
  },
});
