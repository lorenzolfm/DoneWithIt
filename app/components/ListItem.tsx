import React from 'react';
import { GestureResponderEvent, Image, ImageSourcePropType, StyleSheet, TouchableHighlight, View } from 'react-native';
import Swipeable from 'react-native-gesture-handler/Swipeable';

import { AppText } from './AppText';

import colors from '../config/colors';

type ListItemProps = {
  title: string,
  subtitle: string,
  image: ImageSourcePropType,
  onPress?: (event: GestureResponderEvent) => void,
  renderRightActions?: any,
};

export const ListItem = ({ title, subtitle, image, onPress, renderRightActions }: ListItemProps) => {
  return (
    <Swipeable renderRightActions={renderRightActions}>
      <TouchableHighlight
        onPress={onPress}
        underlayColor={colors.light}
      >
        <View style={styles.container}>
          <Image style={styles.image} source={image} />
          <View>
            <AppText style={styles.title}>{title}</AppText>
            <AppText style={styles.subtitle}>{subtitle}</AppText>
          </View>
        </View>
      </TouchableHighlight>
    </Swipeable>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
  },
  image: {
    width: 70,
    height: 70,
    borderRadius: 35,
    marginRight: 10,
  },
  title: {
    fontWeight: 'bold',
  },
  subtitle: {
    color: colors.medium,
  },
});
