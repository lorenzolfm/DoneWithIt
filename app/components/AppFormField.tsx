import React from 'react';
import { useFormikContext } from 'formik';

import { AppTextInput } from './AppTextInput';
import { ErrorMessage } from './ErrorMessage';

type Props = {
  name: string,
  [otherProp: string]: any,
};
export const AppFormField = ({ name, ...otherProps }: Props) => {
  const { setFieldTouched, handleChange, errors, touched } = useFormikContext();

  return (
    <>
      <AppTextInput
        onBlurEvent={() => setFieldTouched(name)}
        onChangeText={handleChange(name)}
        {...otherProps}
      />
      <ErrorMessage error={errors[name]} visible={touched[name]} />
    </>
  );
};
