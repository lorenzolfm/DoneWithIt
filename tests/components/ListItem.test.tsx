import { shallow } from 'enzyme';
import { ListItem } from '../../app/components/ListItem';

describe('ListItem', () => {
  it('matches snapshot', () => {
    expect(
      shallow(
        <ListItem
          title="test-title"
          subtitle="test-subtitle" 
          image={require('../../app/assets/jacket.jpg')}
          />
       )
    ).toMatchSnapshot();
  });
});
