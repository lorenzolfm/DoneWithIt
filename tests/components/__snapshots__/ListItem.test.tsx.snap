// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem matches snapshot 1`] = `
<View>
  <ForwardRef
    underlayColor="#f8f4g4"
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 15,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/jacket.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 35,
            "height": 70,
            "marginRight": 10,
            "width": 70,
          }
        }
      />
      <View>
        <AppText
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          test-title
        </AppText>
        <AppText
          style={
            Object {
              "color": "#6e6969",
            }
          }
        >
          test-subtitle
        </AppText>
      </View>
    </View>
  </ForwardRef>
</View>
`;
